src/checkpoint/dispatch/vrt/derived.h

```diff
@@ -0,0 +1,124 @@
+    ::checkpoint::dispatch::vrt::TypeIdx expected_idx                                \
+  ) override {                                                                       \
+    ::checkpoint::instantiateObjSerializer<                                          \
+      DERIVED,                                                                       \
+      checkpoint_serializer_variatic_args()                                          \
+    >();                                                                             \
+    debug_checkpoint(                                                                \
+      "%s: BEGIN: _checkpointDynamicSerialize: serializer_idx=%d {\n",               \
+      typeid(DERIVED).name(), base_ser_idx                                           \
+    );                                                                               \
+    ::checkpoint::dispatch::vrt::assertTypeIdxMatch<DERIVED>(expected_idx);          \
+    auto base_idx = ::checkpoint::dispatch::vrt::objregistry::makeObjIdx<BASE>();    \
+    BASE::_checkpointDynamicSerialize(s, base_ser_idx, base_idx);                    \
+    auto dispatcher =                                                                \
+      ::checkpoint::dispatch::vrt::serializer_registry::getBaseIdx<DERIVED>(         \
+        base_ser_idx                                                                 \
+      );                                                                             \
+    dispatcher(s, *static_cast<DERIVED*>(this));                                     \
+    debug_checkpoint(                                                                \
+      "%s: END: _checkpointDynamicSerialize: serializer_idx=%d }\n",                 \
+      typeid(DERIVED).name(), base_ser_idx                                           \
+    );                                                                               \
+  }                                                                                  \
+  ::checkpoint::dispatch::vrt::TypeIdx _checkpointDynamicTypeIndex() override {      \
+    return ::checkpoint::dispatch::vrt::DispatchTypeIdx<DERIVED>::get();             \
+  }                                                                                  \
+
+namespace checkpoint { namespace dispatch { namespace vrt {
+
+/**
+ * \brief A derived class of an inheritance hierarchy should inherit from this
+ *
+ * \param DerivedT the derived class, following CRTP
+ * \param BaseT the base class to inherit from and identify the relevant hierarchy
+ *
+ * The serialize() method of such derived classes need not call any
+ * base class serialize() explicitly
+ */
+template <typename DerivedT, typename BaseT>
+struct SerializableDerived : BaseT {
+  template <typename... Args>
+  explicit SerializableDerived(Args&&... args)
+    : BaseT(std::forward<Args>(args)...)
+  { }
+
+  SerializableDerived() = default;
+
+  template <typename SerializerT>
```

Comment:
Do we actually need this?

Reply:
I think not - add an assertion and check that it's never called. It shouldn't be, I don't think.